src/UI/Dropdown.elm

```diff
@@ -0,0 +1,437 @@
+
+-}
+init : String -> State item
+init id =
+    State <| Dropdown.init id
+
+
+{-| Given a message, apply an update to the [`Dropdown.State`](#State).
+Do not ignore the returned `Effect`, it may include remote select's messages.
+
+    ( newModel, newCmd ) =
+        Dropdown.update renderConfig subMsg dropdown
+
+-}
+update : RenderConfig -> Msg item -> Dropdown item msg -> ( State item, Effect msg )
+update cfg (Msg msg) dropdown =
+    let
+        (Dropdown prop _) =
+            dropdown
+
+        (State state) =
+            prop.state
+
+        ( state_, effects ) =
+            Dropdown.updateWithoutPerform (getConfig cfg dropdown) msg () state
+    in
+    ( State state_, List.map translateEffect effects )
+
+
+translateEffect : Dropdown.Effect msg -> Effect.SideEffect msg
+translateEffect effect =
+    case effect of
+        Dropdown.Loopback msg ->
+            Effect.MsgToCmd msg
+
+        Dropdown.DomFocus msg id ->
+            Effect.DomFocus msg id
+
+
+{-| Constructs a basic dropdown.
+Also defines the handling function for messages, and the current dropdown's state.
+
+    basic
+        { dropdownMsg = ForDropdownMsg
+        , onSelectMsg = GotSelectItemMsg
+        , state = model.dropdownState
+        }
+
+-}
+basic : BasicConfig item msg -> Dropdown item msg
+basic prop =
+    Dropdown
+        { dropdownType = Basic
+        , items = []
+        , dropdownMsg = prop.dropdownMsg
+        , onSelectMsg = prop.onSelectMsg
+        , state = prop.state
+        }
+        defaultOptions
+
+
+{-| Constructs a filterable dropdown.
+Also defines the handling function for messages, and the current dropdown's state.
+
+    filterable
+        { dropdownMsg = ForDropdownMsg
+        , onSelectMsg = GotSelectItemMsg
+        , state = model.dropdownState
+        }
+
+-}
+filterable : BasicConfig item msg -> Dropdown item msg
+filterable prop =
+    Dropdown
+        { dropdownType = Filterable
+        , items = []
+        , dropdownMsg = prop.dropdownMsg
+        , onSelectMsg = prop.onSelectMsg
+        , state = prop.state
+        }
+        defaultOptions
+
+
+defaultOptions : Options item
+defaultOptions =
+    { placeholder = Nothing
+    , filterPlaceholder = Nothing
+    , selected = Nothing
+    , itemToText = always ""
+    }
+
+
+{-| Replaces the component's placeholder text.
+
+    Dropdown.withPlaceholder "Choose a book"
+        someDropdown
+
+-}
+withPlaceholder : String -> Dropdown item msg -> Dropdown item msg
+withPlaceholder placeholder (Dropdown prop opt) =
+    Dropdown prop { opt | placeholder = Just placeholder }
+
+
+{-| Replaces the filterable component's placeholder text.
+
+    Dropdown.withPlaceholder "Choose a book"
+        someDropdown
+
+-}
+withFilterPlaceholder : String -> Dropdown item msg -> Dropdown item msg
+withFilterPlaceholder placeholder (Dropdown prop opt) =
+    Dropdown prop { opt | filterPlaceholder = Just placeholder }
+
+
+{-| Replaces the component's list of elements.
```

Comment:
I don't think we should use the word "replaces" unless there's a relevant default value to be considered and even in that case we could use "override", otherwise I suggest writing "change" or "set"

Reply:
> Looks really good, my only other complaint is that the font size used for the input in the filterable version looks too big compared to the rest ("Choose a book" in the image)
> 
> ![image](https://user-images.githubusercontent.com/14297772/122068662-f906ea80-cdca-11eb-847c-00508198f7e1.png)
> 
> I couldn't find an example of the filterable dropdown having a different font size in figma, so I'm assuming it's the same

We'll have to [change `elm-ui-dropdown` to fix that](https://github.com/PaackEng/elm-ui-dropdown/blob/master/src/Dropdown.elm#L680-L681), I think `withFilterPlaceholder` should receive a [`Input.Placeholder`](https://package.elm-lang.org/packages/mdgriffith/elm-ui/latest/Element-Input#Placeholder) instead of a `String`.